packages/common/src/project/models.ts

```diff
@@ -185,7 +185,7 @@ export class CustomDataSourceBase<
   startBlock?: number;
   @Type(() => FileReferenceImpl)
   @ValidateNested({each: true})
-  assets: {[p: string]: CustomDataSourceAsset};
+  assets: Map<string, CustomDataSourceAsset>;
```

Comment:
```suggestion
  assets: Record<string, CustomDataSourceAsset>;
```

Reply:
Doing record/dictionary doesn't seem to be supported https://github.com/typestack/class-transformer/issues/31

I've updated the test for appropriate behaviour. 

Although the assets are never used by the Moonbeam processor so we could probably drop the field altogether